django_tenants/middleware/main.py

```diff
@@ -19,23 +20,45 @@ class TenantMainMiddleware(MiddlewareMixin):
     """
 
     @staticmethod
-    def hostname_from_request(request):
-        """ Extracts hostname from request. Used for custom requests filtering.
-            By default removes the request's port and common prefixes.
+    @lru_cache(maxsize=128)
+    def hostname_from_request_host(request_host):
```

Comment:
Factually your are right, the parameter name and method name do not cover what is going on and could be improved. But, that does not justifies introducing a backwards incompatible change.